### PR TITLE
[Storage] Log test method entry and exit.

### PR DESCRIPTION
**Pull request by kasobol-msft**
- log test start and end with duration (actual times are captured by devops log automagically)
- don't swallow Throwable in interceptor (i.e. don't swallow possible OOMs).

### CODE REVIEW

File: sdk/storage/azure-storage-common/src/test-shared/java/com/azure/storage/common/test/shared/extensions/TestHeaderExtension.java [hunk 1]
```diff
@@ -19,14 +19,17 @@ public void visitSpec(SpecInfo specInfo) {
 
     private static class TestHeaderIterationInterceptor implements IMethodInterceptor {
         @Override
-        public void intercept(IMethodInvocation invocation) {
+        public void intercept(IMethodInvocation invocation) throws Throwable {
             // Print out the test name to create breadcrumbs in our test logging in case anything hangs.
             String testName = TestNameProvider.getTestName(invocation.getIteration());
-            System.out.printf("========================= %s =========================%n", testName);
+            System.out.printf("%s is starting", testName);
```
Comment (alzimmermsft) [inline sdk/storage/azure-storage-common/src/test-shared/java/com/azure/storage/common/test/shared/extensions/TestHeaderExtension.java:25]: Thoughts on switching over to using a logger here given that CI has logging configurations?

Also, add a `%n` to the `printf` so it is new-lined.

File: sdk/storage/azure-storage-common/src/test-shared/java/com/azure/storage/common/test/shared/extensions/TestHeaderExtension.java [hunk 2]
```diff
@@ -19,14 +19,17 @@ public void visitSpec(SpecInfo specInfo) {
...
+            long startTimestamp = System.currentTimeMillis();
+
             try {
                 invocation.proceed();
-            } catch (Throwable throwable) {
-                throwable.printStackTrace();
+            } finally {
```
Comment (alzimmermsft) [inline sdk/storage/azure-storage-common/src/test-shared/java/com/azure/storage/common/test/shared/extensions/TestHeaderExtension.java:30]: Do we want to keep the `catch` and log the testing error?

File: sdk/storage/azure-storage-common/src/test-shared/java/com/azure/storage/common/test/shared/extensions/TestHeaderExtension.java [hunk 3]
```diff
@@ -19,14 +19,17 @@ public void visitSpec(SpecInfo specInfo) {
...
+                long duration = System.currentTimeMillis() - startTimestamp;
```
Comment (alzimmermsft) [inline sdk/storage/azure-storage-common/src/test-shared/java/com/azure/storage/common/test/shared/extensions/TestHeaderExtension.java:31]: Same question on using a log here instead.

Also, add a `%n` to the `printf` so it is new-lined.

**Review by alzimmermsft — APPROVED** [review]

File: sdk/storage/azure-storage-common/src/test-shared/java/com/azure/storage/common/test/shared/extensions/TestHeaderExtension.java (hunk 1 above)
Comment (kasobol-msft) [reply sdk/storage/azure-storage-common/src/test-shared/java/com/azure/storage/common/test/shared/extensions/TestHeaderExtension.java]: I'm going to add logger but keep printf - printf output displays on devops job page which is nice, but having entries in log file are going to help analyze it without crossreferencing with devops output.

File: sdk/storage/azure-storage-common/src/test-shared/java/com/azure/storage/common/test/shared/extensions/TestHeaderExtension.java (hunk 2 above)
Comment (kasobol-msft) [reply sdk/storage/azure-storage-common/src/test-shared/java/com/azure/storage/common/test/shared/extensions/TestHeaderExtension.java:34]: I think we might have disturbed the test framework by intercepting this in such a way.

If it's exception then test framework should dump it, if it's Error then JVM should dump it into output when it bubbles.

**Pull request merged**